dvc/remote/base.py

```diff
@@ -847,9 +847,22 @@ def cache_exists(self, checksums, jobs=None, name=None):
         checksums = frozenset(checksums)
         prefix = "0" * self.TRAVERSE_PREFIX_LEN
         total_prefixes = pow(16, self.TRAVERSE_PREFIX_LEN)
-        remote_checksums = set(
-            map(self.path_to_checksum, self.list_cache_paths(prefix=prefix))
-        )
+        with Tqdm(
+            desc="Estimating size of "
+            + ("cache in '{}'".format(name) if name else "remote cache"),
+            unit="file",
+        ) as pbar:
+            remote_checksums = set(
+                map(
+                    self.path_to_checksum,
+                    self.list_cache_paths(
+                        prefix=prefix,
+                        progress_callback=lambda n=1: pbar.update(
+                            n * total_prefixes
+                        ),
+                    ),
```

Comment:
Would extract this into a separate var and use it instead. There are too many levels of brackets in this set(map(... line :slightly_smiling_face: Same down below.